### ISSUE TITLE
Name collision with deprecated modules causes `deprecated-module` false-positive

### ISSUE THREAD

**Issue by RaenonX** [PyCQA/pylint#4672]
### Steps to reproduce

```python
from .parser import ParserBase
```

There is a file called `parser.py` in the same directory as the file above.

### Current behavior

Emits `deprecated-module`.

### Expected behavior

Should not emit `deprecated-module` since it's not using the deprecated `parser` module.

I assume it's somehow ignoring that `.` before the module name.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.9.3
astroid 2.6.2
Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
Duplicate of #4629
